core/server/middleware/index.js

```diff
@@ -1,196 +1,54 @@
+    spamPrevention   = require('./spam-prevention'),
+    staticTheme      = require('./static-theme'),
+    uncapitalise     = require('./uncapitalise'),
+    oauth            = require('./oauth'),
+
+    themeHandler     = require('./theme-handler'),
+    privateBlogging  = require('./private-blogging'),
 
     blogApp,
+    middleware,
     setupMiddleware;
 
-// ##Custom Middleware
-
-// ### GhostLocals Middleware
-// Expose the standard locals that every external page should have available,
-// separating between the theme and the admin
-function ghostLocals(req, res, next) {
-    // Make sure we have a locals value.
-    res.locals = res.locals || {};
-    res.locals.version = config.ghostVersion;
-    res.locals.safeVersion = config.ghostVersion.match(/^(\d+\.)?(\d+)/)[0];
-    // relative path from the URL
-    res.locals.relativeUrl = req.path;
-
-    next();
-}
-
-// ### Activate Theme
-// Helper for manageAdminAndTheme
-function activateTheme(activeTheme) {
-    var hbsOptions,
-        themePartials = path.join(config.paths.themePath, activeTheme, 'partials');
-
-    // clear the view cache
-    blogApp.cache = {};
-
-    // set view engine
-    hbsOptions = {
-        partialsDir: [config.paths.helperTemplates],
-        onCompile: function onCompile(exhbs, source) {
-            return exhbs.handlebars.compile(source, {preventIndent: true});
-        }
-    };
-
-    fs.stat(themePartials, function stat(err, stats) {
-        // Check that the theme has a partials directory before trying to use it
-        if (!err && stats && stats.isDirectory()) {
-            hbsOptions.partialsDir.push(themePartials);
-        }
-    });
-
-    blogApp.engine('hbs', hbs.express3(hbsOptions));
-
-    // Update user error template
-    errors.updateActiveTheme(activeTheme);
-
-    // Set active theme variable on the express server
-    blogApp.set('activeTheme', activeTheme);
-}
-
-// ### configHbsForContext Middleware
-// Setup handlebars for the current context (admin or theme)
-function configHbsForContext(req, res, next) {
-    var themeData = config.theme;
-    if (req.secure && config.urlSSL) {
-        // For secure requests override .url property with the SSL version
-        themeData = _.clone(themeData);
-        themeData.url = config.urlSSL.replace(/\/$/, '');
-    }
-
-    hbs.updateTemplateOptions({data: {blog: themeData}});
-
-    if (config.paths.themePath && blogApp.get('activeTheme')) {
-        blogApp.set('views', path.join(config.paths.themePath, blogApp.get('activeTheme')));
+middleware = {
+    busboy: busboy,
+    cacheControl: cacheControl,
+    spamPrevention: spamPrevention,
+    privateBlogging: privateBlogging,
+    api: {
+        addClientSecret: clientAuth.addClientSecret,
+        cacheOauthServer: clientAuth.cacheOauthServer,
+        authenticateClient: clientAuth.authenticateClient,
+        generateAccessToken: clientAuth.generateAccessToken,
+        methodNotAllowed: apiErrorHandlers.methodNotAllowed,
```

Comment:
The `methodNotAllowed` function doesn't exist any more - any chance @sebgie or @halfdan could update this in one of the commits? I can change it separately but it'd nice to keep it in this PR. This can otherwise be merged.